src/generators/api-links/utils/extractExports.mjs

```diff
@@ -0,0 +1,215 @@
+            exports.ctors.push(value.name);
+          }
+        });
+
+        break;
+
+      default:
+        exports.identifiers.push(rhs.name);
+        break;
+    }
+  }
+
+  return exports;
+}
+
+/**
+ * @param {import('acorn').VariableDeclarator} declaration
+ * @param {import('acorn').SourceLocation} loc
+ * @param {string} basename
+ * @param {Record<string, number>} nameToLineNumberMap
+ * @returns {import('../types').ProgramExports | undefined}
+ */
+function extractVariableDeclaration(
+  { id, init },
+  loc,
+  basename,
+  nameToLineNumberMap
+) {
+  while (init && init.type === 'AssignmentExpression') {
+    // Move left until we get to what we're assigning to
+    //  (i.e. `a=b`, we want `a`)
+    init = init.left;
+  }
+
+  if (!init || init.type !== 'MemberExpression') {
+    // Doesn't exist or we're not writing to a member (probably a normal var,
+    //  like `const a = 123`)
+    return undefined;
+  }
+
+  /**
+   * @type {import('../types').ProgramExports}
+   */
+  const exports = {
+    ctors: [],
+    identifiers: [],
+  };
+
+  if (init.object.name === 'exports') {
+    // Assigning a property in `module.exports` (i.e. `module.exports.asd = ...`)
+    nameToLineNumberMap[`${basename}.${init.property.name}`] = loc.start.line;
+  } else if (
+    init.object.name === 'module' &&
+    init.property.name === 'exports'
+  ) {
+    // Assigning `module.exports` as a whole, (i.e. `module.exports = {}`)
+    exports.ctors.push(id.name);
+    nameToLineNumberMap[id.name] = loc.start.line;
+  }
+
+  return exports;
+}
+
+/**
+ * We need to find what a source file exports so we know what to include in
+ * the final result. We can do this by going through every statement in the
+ * program looking for assignments to `module.exports`.
+ *
+ * Noteworthy that exports can happen throughout the program so we need to
+ * go through the entire thing.
+ *
+ * @param {import('acorn').Program} program
+ * @param {string} basename
+ * @param {Record<string, number>} nameToLineNumberMap
+ * @returns {import('../types').ProgramExports}
+ */
+export function extractExports(program, basename, nameToLineNumberMap) {
+  /**
+   * @type {import('../types').ProgramExports}
+   */
+  const exports = {
+    ctors: [],
+    identifiers: [],
+  };
+
+  program.body.forEach(statement => {
+    const { loc } = statement;
+    if (!loc) {
+      return;
+    }
+
+    switch (statement.type) {
+      case 'ExpressionStatement': {
+        const { expression } = statement;
+        if (expression.type !== 'AssignmentExpression' || !loc) {
+          break;
+        }
+
+        const expressionExports = extractExpression(
+          expression,
+          loc,
+          basename,
+          nameToLineNumberMap
+        );
+
+        if (expressionExports) {
+          exports.ctors.push(...expressionExports.ctors);
+          exports.identifiers.push(...expressionExports.identifiers);
+        }
+
+        break;
+      }
+
```

Comment:
Not sure how common it is to have empty lines between cases. I think it would be valid to have an inline doc

```
// @see reference to what `VariableDeclaration` is
```